src/uu/ls/src/ls.rs

```diff
@@ -978,12 +979,37 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
     list(locs, Config::from(matches))
 }
 
+struct PathData {
+    md: std::io::Result<Metadata>,
+    path_string: String,
+    name: String,
+    p_buf: PathBuf,
+}
+
+impl PathData {
+    fn new(p_buf: PathBuf, config: &Config) -> Self {
+        let md = get_metadata(&p_buf, config);
+        let path_string = p_buf.to_string_lossy().into_owned();
+        let name = p_buf
+            .file_name()
+            .map_or(String::new(), |s| s.to_string_lossy().into_owned());
+        // let path_name: String =
```

Comment:
please remove this comment

Reply:
Yes - this got left in unintentionally, I've removed it in updated version